### PR TITLE
Add doc for testing IoT Edge daemon API endpoints

### PR DESCRIPTION
**Pull request by philipktlin**

### CODE REVIEW

File: edgelet/README.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 This is the code repository for the IoT Edge Security Daemon.
 The daemon provides an abstraction over Hardware Security Modules (HSMs), enabling secure SAS token and certificate provisioning for IoT Edge modules.
 
+## Test IoT Edge daemon API endpoints on dev machine
```
Comment (myagley) [inline edgelet/README.md]: Can we move this link and description to the `devguide`? I think we should keep this readme clean and simple, and put the more interesting info in the dev guide.

File: edgelet/doc/testiotedgedapi.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+2018-10-11T16:29:03Z [INFO] - [mgmt] - - - [2018-10-11 16:29:03.255889269 UTC] "GET /systeminfo?api-version=2018-06-28 HTTP/1.1" 200 OK 60 "-" "-" pid(any)
+2018-10-11T16:29:03Z [INFO] - [work] - - - [2018-10-11 16:29:03.452050679 UTC] "POST /modules/%24edgeAgent/genid/636748070676263768/decrypt?api-version=2018-06-28 HTTP/1.1" 200 OK 1236 "-" "-" pid(any)
+2018-10-11T16:29:03Z [INFO] - [work] - - - [2018-10-11 16:29:03.543401118 UTC] "POST /modules/%24edgeAgent/genid/636748070676263768/sign?api-version=2018-06-28 HTTP/1.1" 200 OK 57 "-" "-" pid(any)
+2018-10-11T16:29:03Z [INFO] - [mgmt] - - - [2018-10-11 16:29:03.666959495 UTC] "GET /modules?api-version=2018-06-28 HTTP/1.1" 200 OK 1581 "-" "-" pid(any)
+2018-10-11T16:29:04Z [INFO] - [work] - - - [2018-10-11 16:29:04.627120292 UTC] "POST /modules/%24edgeAgent/genid/636748070676263768/encrypt?api-version=2018-06-28 HTTP/1.1" 200 OK 1261 "-" "-" pid(any)
+```
+
+7. You can use `curl` to send request to API ednpoint for testing.  For example, test [list_modules](https://github.com/Azure/iotedge/blob/master/edgelet/management/docs/ModuleApi.md#list_modules) endpoint:
```
Comment (myagley) [inline edgelet/doc/testiotedgedapi.md]: typo:
```suggestion
7. You can use `curl` to send request to API endpoint for testing.  For example, test [list_modules](https://github.com/Azure/iotedge/blob/master/edgelet/management/docs/ModuleApi.md#list_modules) endpoint:
```
